app/Major.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+
+class Major extends Model
+{
+    /**
+     * The attributes that are not mass assignable.
+     *
+     * @var array<string>
+     */
+    protected $guarded = [
+        'id',
+        'deleted_at',
+        'created_at',
+        'updated_at',
+    ];
+
+    /**
+     * The attributes that should be mutated to dates.
+     *
+     * @var array<string>
+     */
+    protected $dates = [
+        'created_at',
+        'updated_at',
+    ];
+
+    /**
+     *  Get the Users that are members of this Major.
+     */
+    public function members(): BelongsToMany
+    {
+        return $this->belongsToMany(User::class)->whereNull('major_user.deleted_at')->withTimestamps();
```

Comment:
Is the `deleted_at` check required?

Reply:
We're soft deleting relationships which laravel doesn't natively support, so yes.